src/managed/Microsoft.DotNet.Build.Bundle/Program.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.DotNet.Build.Bundle
+{
+    /// <summary>
+    ///  The main driver for Bundle and Extract operations.
+    /// </summary>
+    public static class Program
+    {
+        enum RunMode
+        {
+            Help,
+            Bundle,
+            Extract
+        };
+
+        static RunMode Mode = RunMode.Bundle;
+
+        // Common Options:
+        static bool Verbose = false;
+        static string OutputDir;
+
+        // Bundle options:
+        static bool EmbedPDBs = false;
+        static string HostName;
+        static string ContentDir;
+
+        // Extract options:
+        static string BundleToExtract;
+
+        static void Usage()
```

Comment:
@KathleenDollard @richlander 
Can you please review the command line interface of the bundler tool?

The developers are only expected to interact with the bundler via the build-system, as explained [here](https://github.com/dotnet/designs/blob/master/accepted/single-file/design.md#build-system-interface). But this is still a tool available to the public.

Reply:
@swaroop-sridhar I'm sorry that I didn't realize this earlier - you are writing your own command line parsing? Is there a reason you are not using [System.CommandLine](https://github.com/dotnet/command-line-api)? It will allow much easier maintenance and development for you. And provide help a consistent help format. 

I'll do the specific things first, then the general:

- All switches/options should be in Posix form with two dashes (--directory) and words dash separated
- On occasion, where there is a very common switch an abbreviation is fine. But not for all switches.
- Common CLI switches should be used where they appear, along with common abbreviations. Output is an example. 
- -v is --verbosity and has arguments for detailed, normal, etc. It's fine to have multiple verbosity levels display the same thing, but users can predict what to type.
- We use --help and -h.

General: You are using a switch as a command. I think it is confusing to your users (unless I misunderstand your context) and makes it almost impossible to validate user input (free with System.CommandLine). You're not communicating that app host only makes sense on publish. I don't understand the problem space well enough, but it looks like

bundle publish  <publish-dir> --app-host <host-exe> -o <output-dir>
bundle extract <bundle-exe> -o <output-dir>

So your user gets help on the root, which tells them the two things they can do. Once they decide what they want to do, they get help on the right options. (See dotnet CLI help, or git help). 

OK, that sounds like a parsing nightmare, I know. But @jonsequitur or  I can get show you how to build your CLI in under an hour and then you can drop your parsing logic. You'll get help for free, validation if you want it and other features like tab completion if the user installs it. 

I've been out sick, but I should be available most of this week - limited availability next week.